### PR TITLE
add initial invariants list

### PR DESCRIPTION
**Pull request by mattcushman**

### CODE REVIEW

**Comment by mattcushman** [comment]
some additions/edits to the list

File: test/INVARIANTS.md [hunk 1]
```diff
@@ -4,32 +4,37 @@
...
     - **FS1**: Only 10 proposals can be voted on in the funding stage
     - **FS2**: Votes can only be cast on a proposal in it's distribution period's funding stage.
-    - **FS3**: Cumulative votes cast should be less than or equal to the sum of the squares of the votes cast.    
+    - **FS3**: Cumulative votes cast should be less than or equal to the sum of the squares of the votes cast. (Q: I don't follow this)
+    - **FS4**: Sum of square of votes cast by a given actor are less than or equal to the actor's Ajna held+delegated, squared.
 
 - #### Challenge Stage:
     - **CS1**: Funded proposal slate's cumulative tokens requested should be less than or equal to 90% of the GBC.
-    - **CS2**: Funded proposal slate should be never contain a proposal with negative votes.
+    - **CS2**: Funded proposal slate should be never contain a proposal with negative votes. (Q: is this actually enforced in the code??)
```
Comment (MikeHathaway) [inline test/INVARIANTS.md:26]: Yes, it's enforced in `checkSlate` line 321 in StandardFunding.sol. `if (proposal.fundingVotesReceived < 0) return false;`

File: test/INVARIANTS.md [hunk 2]
```diff
@@ -4,32 +4,37 @@
...
     - **SS4**: Vote tally for a proposal can only be positive.
     - **SS5**: Votes can only be cast on a proposal in it's distribution period's screening stage.
+    - **SS6**: For every proposal, it is included in the top 10 list if, and only if, it has as many or more votes as the last member of the top ten list (typically the 10th of course, but it may be shorter than ten proposals).
 
 
 - #### Funding Stage:
     - **FS1**: Only 10 proposals can be voted on in the funding stage
     - **FS2**: Votes can only be cast on a proposal in it's distribution period's funding stage.
-    - **FS3**: Cumulative votes cast should be less than or equal to the sum of the squares of the votes cast.    
+    - **FS3**: Cumulative votes cast should be less than or equal to the sum of the squares of the votes cast. (Q: I don't follow this)
```
Comment (MikeHathaway) [inline test/INVARIANTS.md:21]: I was attempting to state what you stated in FS4 more clearly. I'd favor replacing my FS3 with your FS4

File: test/INVARIANTS.md [hunk 3]
```diff
@@ -4,32 +4,37 @@
 
 - #### Distribution Period:
     - **DP1**: Only one distribution period should be active at a time
+    - **DP2**: Each winning proposal successfully claims no more that what was finalized in the challenge stage
```
Comment (MikeHathaway) [inline test/INVARIANTS.md:7]: We should also add an invariant around the size of the GBC across many distribution periods. Specifically, there could be a case where a distribution period ends, proposals can be but are not executed, and several more distribution periods start. The treasury should take into account funds held for execution by finalized proposals. We handle this logic in `StandardFunding.sol:startNewDistributionPeriod()`

**Review by MikeHathaway — APPROVED** [review]
LGTM

**Pull request merged**